das/checkpoint_store.go

```diff
@@ -0,0 +1,45 @@
+package das
+
+import (
+	"encoding/binary"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+)
+
+var (
+	storePrefix   = datastore.NewKey("das")
+	checkpointKey = datastore.NewKey("checkpoint")
+)
+
+// NewCheckpointStore wraps the given datastore.Datastore with
+// the `das` prefix.
+func NewCheckpointStore(ds datastore.Datastore) datastore.Datastore {
+	return namespace.Wrap(ds, storePrefix)
+
+}
+
+// loadCheckpoint loads the DAS checkpoint from disk and returns it.
+// If there is no known checkpoint, it returns height 0.
+func loadCheckpoint(ds datastore.Datastore) (int64, error) {
```

Comment:
This function and the next one assume a prefixed store right? Is there a way to enforce this at the type level? Otherwise it's relying on the library user knowing to use a prefix store.

Reply:
I could make a checkpoint store type, yes

Reply:
Actually I don't think this is necessary - will push a better fix for this.